### PR TITLE
Nix Environment and Simulator Build

### PR DESCRIPTION
**Pull request by HU90m**
```
# Create a shell in which one can build the simulator
nix develop
# Build the simulator in a reproducible environment
nix build .#sonata_simulator
```

### CODE REVIEW

**Review by marnovandermaas — COMMENTED** [review]
Do these files need to live in the root of the repository or can we put them in a `nix` directory?

File: flake.nix [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      pkgs = import nixpkgs {
+        inherit system;
+      };
+      lr_pkgs = lowrisc-nix.outputs.packages.${system};
+
+      sim_build_deps =
+        (with pkgs; [libelf zlib python311Packages.pip])
+        ++ (with lr_pkgs; [python_ot verilator_ot]);
+
+      sonata_simulator = pkgs.stdenv.mkDerivation {
```
Comment (nbdd0121) [inline flake.nix:26]: Installables usually use dash instead of underscore.

File: flake.nix [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        pname = "sonata-system-simulator";
+        version = sonata_version;
+        src = builtins.path {path = ./.;};
```
Comment (nbdd0121) [inline flake.nix:29]: Does 
```suggestion
        src = ./.;
```
not work?

File: flake.nix [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        nativeBuildInputs = sim_build_deps;
```
Comment (nbdd0121) [inline flake.nix:30]: Libraries should live in buildInputs not nativeBuildInputs

File: flake.nix [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        buildPhase = ''
+          export HOME=$(mktemp -d)
+          fusesoc --cores-root=. run --target=sim --tool=verilator --setup \
+            --build lowrisc:sonata:system
```
Comment (nbdd0121) [inline flake.nix:34]:
```suggestion
          HOME=$TMPDIR fusesoc --cores-root=. run --target=sim --tool=verilator --setup \
            --build lowrisc:sonata:system
```
perhaps? This won't change people's home when they run buildPhase directly in a dev shell.

**Pull request merged**